libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -36,6 +38,8 @@ use crate::transport::Transport;
 
 const DEFAULT_HEARTBEAT_INTERVAL: u64 = 10;
 const CHANNEL_CAPACITY: usize = 15;
+const INITIAL_RETRY_FREQUENCY: u64 = 10;
```

Comment:
10 seconds seems kinda  high for the initial reconnect. You could probably start lower since it grows exponentially (say 1 to 2 seconds).

Reply:
The reconnection attempts are driven by heartbeats, so it does not make sense to have a lower initial retry frequency then the default heartbeat interval, which is also set to 10. Also the first reconnection attempt will happen immediately when the disconnection is discovered.